Signal/src/ViewControllers/ConversationView/Cells/OWSMessageFooterView.m

```diff
@@ -0,0 +1,185 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWSMessageFooterView.h"
+#import "DateUtil.h"
+#import "Signal-Swift.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface OWSMessageFooterView ()
+
+@property (nonatomic) UILabel *timestampLabel;
+@property (nonatomic) UIView *spacerView;
+@property (nonatomic) UILabel *statusLabel;
+@property (nonatomic) UIView *statusIndicatorView;
+
+@end
+
+@implementation OWSMessageFooterView
+
+// `[UIView init]` invokes `[self initWithFrame:...]`.
+- (instancetype)initWithFrame:(CGRect)frame
+{
+    if (self = [super initWithFrame:frame]) {
+        [self commontInit];
+    }
+
+    return self;
+}
+
+- (void)commontInit
+{
+    // Ensure only called once.
+    OWSAssert(!self.timestampLabel);
+
+    self.layoutMargins = UIEdgeInsetsZero;
+
+    self.axis = UILayoutConstraintAxisHorizontal;
+    self.spacing = self.hSpacing;
+    self.alignment = UIStackViewAlignmentCenter;
+
+    self.timestampLabel = [UILabel new];
+    // TODO: Color
+    self.timestampLabel.textColor = [UIColor lightGrayColor];
+
+    self.spacerView = [UIView new];
+    [self.spacerView setContentHuggingLow];
+
+    self.statusLabel = [UILabel new];
+    // TODO: Color
+    self.statusLabel.textColor = [UIColor lightGrayColor];
+
+    self.statusIndicatorView = [UIView new];
+    [self.statusIndicatorView autoSetDimension:ALDimensionWidth toSize:self.statusIndicatorSize];
+    [self.statusIndicatorView autoSetDimension:ALDimensionHeight toSize:self.statusIndicatorSize];
+    self.statusIndicatorView.layer.cornerRadius = self.statusIndicatorSize * 0.5f;
+}
+
+- (void)configureFonts
+{
+    self.timestampLabel.font = UIFont.ows_dynamicTypeCaption2Font;
+    self.statusLabel.font = UIFont.ows_dynamicTypeCaption2Font;
+}
+
+- (CGFloat)statusIndicatorSize
+{
+    // TODO: Review constant.
+    return 12.f;
+}
+
+- (CGFloat)hSpacing
+{
+    // TODO: Review constant.
+    return 8.f;
+}
+
+#pragma mark - Load
+
+- (void)configureWithConversationViewItem:(ConversationViewItem *)viewItem
+{
+    OWSAssert(viewItem);
+
+    [self configureLabelsWithConversationViewItem:viewItem];
+
+    // TODO:
+    self.statusIndicatorView.backgroundColor = [UIColor orangeColor];
+
+    for (UIView *subview in @[
+             self.timestampLabel,
+             self.statusLabel,
+             self.statusIndicatorView,
+         ]) {
+        [subview removeFromSuperview];
+    }
+    if (viewItem.interaction.interactionType == OWSInteractionType_OutgoingMessage) {
```

Comment:
Would it be cleaner instead of removing and adding the subviews here in `configureWith...` to add all of them only once in `commonInit` and only touch their isHidden property in this method?

Reply:
Done.